app/common/data/interfaces/user.py

```diff
@@ -159,6 +193,42 @@ def remove_platform_admin_role_from_user(user: User) -> None:
     db.session.expire(user)
 
 
+@flush_and_rollback_on_exceptions
+def add_permissions_to_user(
+    user: User,
+    permissions: list[RoleEnum],
+    organisation_id: uuid.UUID,
+    grant_id: uuid.UUID | None = None,
+) -> UserRole:
+    user_role = get_user_role(user, organisation_id, grant_id)
+    existing_permissions = user_role.permissions if user_role else []
+    combined_permissions = list(set(existing_permissions + permissions))
+
+    return upsert_user_role(user, combined_permissions, organisation_id, grant_id)
+
+
+@flush_and_rollback_on_exceptions
+def remove_permissions_from_user(
+    user: User,
+    permissions: list[RoleEnum],
+    organisation_id: uuid.UUID,
+    grant_id: uuid.UUID | None = None,
+) -> UserRole | None:
+    user_role = get_user_role(user, organisation_id, grant_id)
+    if not user_role:
+        return None
+
+    existing_permissions = user_role.permissions
+    combined_permissions = list(set(existing_permissions) - set(permissions))
+
+    if not combined_permissions:
+        db.session.delete(user_role)
```

Comment:
I don't know if we ever made a decision about this but reminder to myself there was a suggestion about removing the "member" role from both access and deliver permissions set ups which might change what the implication of having no permissions means (you still have access but no elevated perms). Unless we do that separately this feels good for keeping the user role table tidy